src/components/Input/Input.tsx

```diff
@@ -0,0 +1,49 @@
+import * as S from './Input.style';
+import { useState } from 'react';
+import { IconEyeOff, IconEyeOn } from '@/public/svgs';
+
+interface Props {
+  placeholder?: string;
+  errorMessage?: string;
+  type?: string;
+  onBlur?: () => void;
+}
+
+function Input({
+  placeholder = '내용 입력',
+  errorMessage,
+  type: initialType = 'text',
+  onBlur,
+}: Props) {
+  const [type, setType] = useState(initialType);
+
+  const toggleShow = () => {
+    setType((prevType) => {
+      if (prevType === 'text') {
+        return 'password';
+      } else {
+        return 'text';
+      }
```

Comment:
이렇게 경우의 수가 제한적인 부분에서는 별도의 코드블럭 및 리턴문보다 삼항연산자를 활용하시는 것도 좋아보이네요

```suggestion
   setType(prevType => prevType === 'text' ? 'password' : 'text');
```